LabBuilder/LabBuilder.psm1

```diff
@@ -4762,6 +4880,200 @@ Function Uninstall-Lab {
 } # Uninstall-Lab
 ####################################################################################################
 
+
+##########################################################################################
+#   Create Template VHD files for Labs
+##########################################################################################
+<#.Synopsis
+	Function used to create Template VHDs from source ISO file - Tested againist Win10/Win Server 2016 TP4
+	Uses Convert-WindowsImage from TP4 CD 
+.DESCRIPTION
+	Uses an XML as the source for some config options and a hashtable of Disk configurations
+.EXAMPLE
+	Initialize-TemplateVHD -config Config.xml -VMTemplateDisk $VMTemplateDisks  
+
+#>
+function Initialize-TemplateVHD
+{
+   param (
+        [Parameter(
+            Mandatory,
+            Position=0)]
+        [XML] $Config,
+
+        [Parameter(
+            Mandatory,
+            Position=1)]
+	   [System.Collections.Hashtable[]] $VMTemplateDisks
+    )
+
+
+	$ScriptDir = $Config.labbuilderconfig.settings.FullConfigPath
+	$TemplateDiskPath = $Config.labbuilderconfig.settings.TemplateDiskPath
+	$ServerISO = $Config.labbuilderconfig.settings.TemplateISO
+
+	[String]$WorkFolder = Join-Path -Path $ScriptDir -ChildPath 'WinImage'
+	[String]$DismFolder = Join-Path -Path $ScriptDir -ChildPath 'DISM'
+	[String]$MountFolder = Join-Path -Path $ScriptDir -ChildPath 'Mount'
+
+	Write-Verbose "Script Directory = $ScriptDir"
+	Write-Verbose "Template disk path = $TemplateDiskpath"
+	Write-Verbose "ISO to use = $ServerISO"
+
+
+	##########################################################################################
+	#   Create Parent VHDX files for VMs
+	##########################################################################################
+	# Mount the Windows Server 2016 ISO and get the drive letter
+
+
+	Write-Verbose -Message 'Mounting Server ISO'
+	$null = Mount-DiskImage -ImagePath $ServerISO
+	[String]$DriveLetter = (Get-Diskimage -ImagePath $ServerISO | Get-Volume).DriveLetter
+	$ISODrive = "$([string]$DriveLetter):"
+
+	Write-Verbose "CD Drive is $ISODrive"
+	Get-PSDrive -PSProvider FileSystem # Work around an issue with script not seeing the drive 
+
+	Write-Verbose 'Copying DISM from Server ISO to Working Folders'
+	If (-not (Test-Path -Path $DismFolder -PathType Container)) 
+		{
+		$null = New-Item -Path $DismFolder -ItemType Directory
+		} #EndIf
+	$null = Copy-Item -Path "$ISODrive\Sources\api*downlevel*.dll" -Destination $DismFolder -Force
+	$null = Copy-Item -Path "$ISODrive\Sources\*dism*" -Destination $DismFolder -Force
+	$null = Copy-Item -Path "$ISODrive\Sources\*provider*" -Destination $DismFolder -Force
+	
+	
+
+	 # As of 2015-06-16 Convert-WindowsImage contains a function instead of being a standalone script.
+	 # . source the Convert-WindowsImage.ps1 so it can be called
+	. "$ScriptDir\Convert-WindowsImage.ps1"
```

Comment:
I wonder if at a later date we could automate the download and install of this script?

If this script was available as a module in PS Gallery it would be really easy.

I'll raise this as an enhancement issue.